src/android/java/io/jxcore/node/BtConnectorHelper.java

```diff
@@ -49,18 +55,20 @@ public BtConnectorHelper() {
         this.context = jxcore.activity.getBaseContext();
     }
 
-
-    public BTConnector.WifiBtStatus Start(String peerName,int port){
+    public synchronized BTConnector.WifiBtStatus Start(String peerName,int port){
         this.mServerPort = port;
         this.myPeerIdentifier= GetBluetoothAddress();
         this.myPeerName = peerName;
-        this.lastAvailableList.clear();
+        synchronized (lastAvailableList) {
+            this.lastAvailableList.clear();
+        }
         Stop();
+
         mBTConnector = new BTConnector(context,this,this,conSettings);
         return mBTConnector.Start(this.myPeerIdentifier,this.myPeerName);
     }
 
-    public boolean isRunning(){
+    public synchronized boolean isRunning(){
```

Comment:
@DrJukka Why is this method synchronized? At worst it can return an outdated response but that is always possible.